### ISSUE TITLE
Set timeout in formatter events

### ISSUE THREAD

**Issue by szikszail** [cucumber/cucumber-js#641]
Hi,

I have implemented a custom formatter for cucumber, what pushes the results to a service. But it might take more time than the previously set default timeout. I know that when we use `registerHandler` method, we can set other timeout, but what is the case with Event functions:

```javascript
const formatter = require('cucumber').Listener.Fomatter({});
formatter.handleAfterFeaturesEvent = function () {
  // some action which definitelly takes more time than defaultTimeout
};
```

Is there any way how I can modify timeout in this case?

**Comment by charlierudolph** [comment]
When creating a listener/formatter you can pass in an options with a timeout in order to specify the timeout for all actions. There is no current way to specify the timeout for indiviual actions. 
```js
const formatter = require('cucumber').Listener.Fomatter({timeout: 5 * 60 * 1000});
```
This isn't documented as custom formatters aren't documented, but I will be adding better support and documentation for them soon.

**Comment by szikszail** [comment]
@charlierudolph does this configuration option work?
I have check the source of formatter.js and didn't found where timeout passed to listener where it's used.

**Comment by charlierudolph** [comment]
Auhh okay. Looks like `options` would need to be passed into Listener for this support to be added.

**Comment by szikszail** [comment]
And do you plan to fix this and add support in the near future?

**Comment by charlierudolph** [comment]
I'm working on 2.0 now which will include docs on custom formatters. That probably won't be out for a month or so. If you'd like this earlier and could make a PR just updating the formatter to pass its options into the listener constructor on the line you pointed out, I'm happy to merge it and release a new version for you.

**Comment by szikszail** [comment]
It would be very cool to include this in the next release! Thank you!

**Comment by charlierudolph** [comment]
Fixed in just released 1.3.1

**Comment by szikszail** [comment]
Thank you!
